Cargo.lock

```diff
@@ -7520,6 +7522,17 @@ dependencies = [
  "unsigned-varint 0.7.2",
 ]
 
+[[package]]
+name = "n-functor"
```

Comment:
What do we know about this new dependency? Doesn't look like it is widely known/used. Does the benefit from using it justify the extra cost/risk?